tools/cbmc/proofs/make_type_header_files.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python3
```

Comment:
This desperately needs documentation.  How is this different from using the C preprocessor to dump the macros?

Reply:
This script doesn't only dump the macros but also includes type declarations extracted from a previously with goto-cc compiled binary. After reading about the C preprocessor flag for gcc, it might be possible to invoke the C preprocessor of some compilers for gathering the macros and merge this with the output of the -dump-c-type-header flag. Using python to grep for the macros makes the script c preprocessor independent.

Reply:
I've added documentation in the `epilog()` function below, thanks!